packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -22,37 +24,50 @@ import 'package:webdev/src/serve/debugger/devtools.dart';
 import 'package:webdev/src/serve/handlers/dev_handler.dart';
 import 'package:webdev/src/serve/server_manager.dart';
 import 'package:webdev/src/serve/webdev_server.dart';
-import 'package:webkit_inspection_protocol/webkit_inspection_protocol.dart' hide StackTrace;
 
 import 'application_package.dart';
 import 'artifacts.dart';
 import 'base/common.dart';
 import 'base/file_system.dart';
 import 'base/logger.dart';
+import 'base/os.dart';
 import 'base/terminal.dart';
 import 'base/utils.dart';
 import 'build_info.dart';
+import 'cache.dart';
 import 'convert.dart';
 import 'device.dart';
 import 'globals.dart';
 import 'project.dart';
 import 'resident_runner.dart';
 import 'run_hot.dart';
+
 import 'web/asset_server.dart';
 
-Future<BuildDaemonClient> connectClient(String workingDirectory, List<String> options, Function(ServerLog) logHandler) {
+Future<BuildDaemonClient> connectClient(
+  String workingDirectory,
+  List<String> options,
+  Function(ServerLog) logHandler,
+) {
+  final String flutterToolsPackages = fs.path.join(Cache.flutterRoot, 'packages', 'flutter_tools', '.packages');
+  final String buildScript = fs.path.join(Cache.flutterRoot, 'packages', 'flutter_tools', 'lib', 'src', 'build_runner', 'build_script.dart');
+  final String flutterWebSdk = artifacts.getArtifactPath(Artifact.flutterWebSdk);
   return BuildDaemonClient.connect(
     workingDirectory,
     // On Windows we need to call the snapshot directly otherwise
     // the process will start in a disjoint cmd without access to
-    // STDIO. We also want to ensure the version of pub is consistent with
-    // the SDK that was used to launch webdev.
+    // STDIO.
     <String>[
       artifacts.getArtifactPath(Artifact.engineDartBinary),
-      fs.path.join(artifacts.getArtifactPath(Artifact.engineDartSdkPath), 'bin', 'pub'),
-      'run',
-      'build_runner',
+      '--packages=$flutterToolsPackages',
+      buildScript,
       'daemon',
+      '--skip-build-script-check',
+      '--delete-conflicting-outputs',
```

Comment:
If there aren't any build to source builders, then we don't need this flag. It probably doesn't hurt though.